packages/autocomplete-js/src/__tests__/detachedMediaQuery.test.ts

```diff
@@ -1,3 +1,71 @@
+import { autocomplete } from '../autocomplete';
+
 describe('detachedMediaQuery', () => {
-  test.todo('tests');
+  const originalMatchMedia = window.matchMedia;
+  const addEventListener = jest.fn();
+
+  beforeAll(() => {
+    Object.defineProperty(window, 'matchMedia', {
+      writable: true,
+      value: jest.fn((query) => ({
+        matches: true,
+        media: query,
+        onchange: null,
+        addListener: jest.fn(),
+        removeListener: jest.fn(),
+        addEventListener,
+        removeEventListener: jest.fn(),
+        dispatchEvent: jest.fn(),
+      })),
+    });
```

Comment:
We could have a constants file in our Jest setup to make sure to set and reset the [`matchMedia`](https://github.com/algolia/autocomplete/blob/next/scripts/jest/setupTests.ts#L7-L19) value to its exact original value:

```tsx
// /scripts/jest/constants.ts
export const matchMedia = jest.fn((query) => ({
  matches: false,
  media: query,
  onchange: null,
  addListener: jest.fn(),
  removeListener: jest.fn(),
  addEventListener: jest.fn(),
  removeEventListener: jest.fn(),
  dispatchEvent: jest.fn(),
}));
```

Wdyt?

Reply:
Make sense since we use it multiple tests. I've also added the `props` so we can provide our mocked functions. (I thought it was a better fit for `test/utils`) 


See: https://github.com/algolia/autocomplete/pull/600/commits/ac0e92c3dea6cf1fabd9371b86584846e0c19e9e